### PR TITLE
Allow for the configuration of max request retries and min/max retry delays in the matrix federation client

### PR DESCRIPTION
**Pull request by MatMaul**
This is a revival of https://github.com/matrix-org/synapse/pull/12504.

It is now using `get_duration`, and the config var names have the unit in their name.

### Pull Request Checklist

* [x] Pull request is based on the develop branch
* [x] Pull request includes a [changelog file](https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#changelog).
* [x] Pull request includes a [sign off](https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#sign-off)
* [x] [Code style](https://matrix-org.github.io/synapse/latest/code_style.html) is correct
  (run the [linters](https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#run-the-linters))

### CODE REVIEW

File: synapse/http/matrixfederationclient.py [hunk 1]
```diff
@@ -406,7 +404,14 @@ def __init__(
         self.clock = hs.get_clock()
         self._store = hs.get_datastores().main
         self.version_string_bytes = hs.version_string.encode("ascii")
-        self.default_timeout = 60
+        self.default_timeout = hs.config.federation.client_timeout_ms / 1000
+
+        self.max_long_retry_delay = hs.config.federation.max_long_retry_delay_ms / 1000
```
Comment (MadLittleMods) [inline synapse/http/matrixfederationclient.py:409]: I feel like we shouldn't be doing the `/ 1000` and then doing `delay * 1000` again below.

Comment (MadLittleMods) [inline synapse/http/matrixfederationclient.py:409]: In any case, let's label the units

```suggestion
        self.max_long_retry_delay_seconds = hs.config.federation.max_long_retry_delay_ms / 1000
```

File: synapse/http/matrixfederationclient.py [hunk 2]
```diff
@@ -730,12 +735,12 @@ async def _send_request(
 
                     if retries_left and not timeout:
                         if long_retries:
-                            delay = 4 ** (MAX_LONG_RETRIES + 1 - retries_left)
-                            delay = min(delay, 60)
+                            delay = 4 ** (self.max_long_retries + 1 - retries_left)
```
Comment (MadLittleMods) [inline synapse/http/matrixfederationclient.py:738]: Add units, `delay_seconds` or `delay_ms`

File: synapse/http/matrixfederationclient.py (hunk 1 above)
Comment (MatMaul) [reply synapse/http/matrixfederationclient.py:409]: I can keep in milli but then I need to add 2 times `* 1000` on L738 and L742 so not sure it's a win.
Also it's consistent with `client_timeout` unit wise, and keep changes to a minimum.

Comment (MatMaul) [reply synapse/http/matrixfederationclient.py]: Another solution is to convert to seconds directly at the config layer.

Comment (MadLittleMods) [reply synapse/http/matrixfederationclient.py]: :shrug: Up to you, much easier to understand with units :fast_forward:

**Pull request merged**